pkg/storage/factory_test.go

```diff
@@ -45,6 +50,60 @@ func TestFactoryStop(t *testing.T) {
 	store.Stop()
 }
 
+func TestNamedStores(t *testing.T) {
+	tempDir := t.TempDir()
+
+	// config for BoltDB Shipper
+	boltdbShipperConfig := shipper.Config{}
+	flagext.DefaultValues(&boltdbShipperConfig)
+	boltdbShipperConfig.ActiveIndexDirectory = path.Join(tempDir, "index")
+	boltdbShipperConfig.SharedStoreType = "filesystem.named-store"
+	boltdbShipperConfig.CacheLocation = path.Join(tempDir, "boltdb-shipper-cache")
+	boltdbShipperConfig.Mode = indexshipper.ModeReadWrite
+
+	cfg := Config{
+		NamedStores: NamedStores{
+			Filesystem: map[string]local.FSConfig{
+				"named-store": {Directory: path.Join(tempDir, "chunks")},
+			},
+		},
+		BoltDBShipperConfig: boltdbShipperConfig,
+	}
+
+	schemaConfig := config.SchemaConfig{
+		Configs: []config.PeriodConfig{
+			{
+				From:       config.DayTime{Time: timeToModelTime(parseDate("2019-01-01"))},
+				IndexType:  "boltdb-shipper",
+				ObjectType: "filesystem.named-store",
+				Schema:     "v9",
+				IndexTables: config.PeriodicTableConfig{
+					Prefix: "index_",
+					Period: time.Hour * 168,
+				},
+			},
+		},
+	}
+
+	limits, err := validation.NewOverrides(validation.Limits{}, nil)
+	require.NoError(t, err)
+
+	t.Run("refer to named store", func(t *testing.T) {
```

Comment:
Not sure about this test name and the following. Maybe 
```suggestion
	t.Run("period config referring to configured named store", func(t *testing.T) {
```
wdyt?

Reply:
👍